lib/rubocop/cop/lint/unified_integer.rb

```diff
@@ -0,0 +1,38 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for using Fixnum or Bignum constant.
+      #
+      # @example
+      #   # bad
+      #   1.is_a?(Fixnum)
+      #   1.is_a?(Bignum)
+      #
+      #   # good
+      #   1.is_a?(Integer)
+      class UnifiedInteger < Cop
```

Comment:
I don't think so. 

In many cases, using `Fixnum` is a mistake in Ruby 2.3 or older too. Then we can use `Integer` instead of `Fixnum`. 
I've found the mistake in some projects. 

e.g.

```ruby
# This statement does nothing when x is a Bignum.
case x
when Fixnum # it's should be `Integer`
  do_something
when Float
  do_something2
end
```

So, I think this cop shoud be enabled for all Ruby versions. It is useful to detect the mistake, and migration to Ruby 2.4.